igvm/vm.py

```diff
@@ -849,3 +866,81 @@ def vm_performance_value(self) -> float:
         ) * hv_perffactor_src
 
         return float(vm_performance_value)
+
+    def aws_get_instances_overview(
+            self, timeout: int = 5) -> Union[List, None]:
+        """AWS Get Instances Overview
+
+        Load or download the latest instances.json, which contains
+        a complete overview about all instance_types, their configuration,
+        performance and pricing.
+
+        :param: timeout: Timeout value for the get request
+
+        :return: VM types overview as list
+                 or None, if load/download failed
+        """
+
+        url = AWS_INSTANCES_OVERVIEW_URL
+        file = Path(AWS_INSTANCES_OVERVIEW_FILE)
+
+        if file.exists():
```

Comment:
what if the content of the file changes? we will never try to re-download it right?

Reply:
Additionally to this PR there will be a new cronjob on our controls, which is doing a daily download of this file

Reply:
i would not add more complexity to this. we really don't need a cronjob. instead we can make use of the http protocol and it's cache headers. just try a `curl -I https://www.ec2instances.info/instances.json`. you will see an `etag` header which is kind of a fingerprint. so i would do it the following way:
- if file does not exist, make a GET request. save the etag from the headers into a separate file, maybe AWS_INSTANCES_OVERVIEW_FILE.etag.
- if file does exist, make a HEAD request and compare the etags. if they differ, repeat the first step. in case the HEAD request fails you can decide whether you want to abort or just ignore and use the existing file in any case (i would still log it so we have a chance to detect problems)

Reply:
> Additionally to this PR there will be a new cronjob on our controls, which is doing a daily download of this file

Igvm can't be dependent on control servers or cron jobs. It's a tool on it's own and must be working fine from peoples' laptops. Please download files when it is run, just like we do with images.

Reply:
The download, if the file was not there, was already in, but I reworked the function, like @Woellchen suggested.

Reply:
cool, the functionality is now good. except for small remarks. see my other comments for that